homepage/tests.py

```diff
@@ -191,3 +191,111 @@ def test_print_details(capsys, review_id, expected_review_details):
     Review.objects.get(pk=review_id).print_details()
 
     assert capsys.readouterr().out == expected_review_details
+
+# -----------prerequisites tests----------- #
+
+
+# three courses:
+@pytest.fixture
+def courses():
+    courses_list = [
+         Course(1, "Course1", True, 3, "N/A", 1.3, 4, 5, 1),
+         Course(2, "Course2", False, 2, "N/A", 3.5, 2, 13, 5),
+         Course(3, "Course3", False, 1, "N/A", 2, 2.5, 0, 0)]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+# creating a new prerequisite log
+@pytest.mark.django_db
+def test_create_new_prerequisite(courses):
+    valid = True
+    preq = Prerequisites(course_id=courses[0], req_course_id=courses[1], req_code=Prerequisites.Req_Code.BEFORE)
+    try:
+        preq.clean_fields()
+        preq.save()
+    except ValidationError:
+        valid = False
+
+    assert valid and Prerequisites.objects.filter(course_id=preq.course_id).count() == 1

```

Comment:
I think you can reuse the method get_prerequisites_for_course
```suggestion
    assert valid and Prerequisites.get_prerequisites_for_course(preq.course_id).count() == 1
```

Reply:
but it's before I test the function itself, this is a more basic test so it should come first, no?

Reply:
I think it should be OK because it's only calling to a method that doing the same as you have here. 
but if you want to leave it like that it's ok by me :)